### ISSUE TITLE
Password/PIN Security

### ISSUE THREAD

**Issue by hank** [Litecoin-Java/bitcoin-wallet#22]
Either on app entrance or on send - on send might be easier.  Thanks to CryptoDude for the suggestion!

**Comment by santa4nt** [comment]
Are the private keys themselves encrypted at rest? If not, maybe the password/PIN can be used as a key to do so. That way, the password/PIN will always be needed every time a private key is involved.

**Comment by ss44** [comment]
https://code.google.com/p/android-lockpattern/wiki/Encryption

Would this be an acceptable library to use? 

I was hoping to try and help implement this, and was looking at a few existing pin / password prompts, this one caught my eye, (largely because I'm a fan of the pattern locks).

**Comment by hank** [comment]
Yeah that would be acceptable if it used encryption packaged in SpongyCastle, which is currently packaged in the app.
